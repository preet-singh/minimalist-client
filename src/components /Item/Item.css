
.Item {
    --note-height: 88px;
    background-color: #00264d;
    color: #cacaca;
    height: 70px;
    margin-bottom: 8px;
    max-width: 300px;
    padding: 8px 0 8px 8px;
    position: relative;
  }
  
  .Item_delete {
    color: #ff0000;
    position: absolute;
    bottom: 0;
    background-color:#00264d;
    border: 0;
    font-size: 15px;
  }

  .Item_delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  
  .Item_title {
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0;
    max-width: 80%;
    min-height: 24px;
  }
  .Item_title a {
    color: #cacaca;
    display: block;
    text-decoration: none;
  }
  .Item_title a:hover {
    color: #FFB72F;
  }

  .Item_edit {
    color: #00b300;
    float: right;
    position: absolute;
    background-color:#00264d;
    border: 0;
    font-size: 15px;
  }  