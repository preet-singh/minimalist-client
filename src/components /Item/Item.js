import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import InventoryContext from '../../InventoryContext';
import config from '../../config';
import './Item.css';
import PropTypes from 'prop-types';

export default class Item extends React.Component {
  static defaultProps ={
    onDeleteItem: () => {},
    onUpdateItem: () => {},
  }
  static contextType = InventoryContext;

  handleClickDelete = e => {
    e.preventDefault()
    const itemId = this.props.id

    fetch(`${config.API_ENDPOINT}/items/${itemId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
      })
      .then(() => {
        this.context.deleteItem(itemId)
        // allow parent to perform extra behaviour
        this.props.onDeleteItem(itemId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  handleClickUpdate = (e, newItemFields) => {
    e.preventDefault()
    const itemId = this.props.id

    fetch(`${config.API_ENDPOINT}/items/${itemId}`, {
      method: 'PATCH',
      headers: {
        'content-type': 'application/json'
      }
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
          else return res.json()
      })
      .then((data) => {
        newItemFields(data)
        this.context.updateItem(data)
        this.props.onUpdateItem(data)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { name, id } = this.props
    return (
      <div className='Item'>
        <h2 className='Item_title'>
          <Link to={`/item/${id}`}>
            {name}
          </Link>
        </h2>
        <button
          className='Item_delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          <FontAwesomeIcon icon='trash-alt' />
          {' '}
        </button>
        <button
          className='Item_edit'
          type='button'
          onClick={this.handleClickUpdate}
        >
          <FontAwesomeIcon icon='edit' />
          {' '}
        </button>
      </div>
    )
  }
}

Item.propTypes = {
  onDeleteItem: PropTypes.func.isRequired
}