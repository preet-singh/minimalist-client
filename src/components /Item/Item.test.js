import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Item from './Item'

describe(`Item component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
  }

  it('renders a .Item by default', () => {
    const wrapper = shallow(<Item />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Item given props', () => {
    const wrapper = shallow(<Item {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
