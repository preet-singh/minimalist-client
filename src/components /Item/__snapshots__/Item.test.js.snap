// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders a .Item by default 1`] = `
<div
  className="Item"
>
  <h2
    className="Item_title"
  >
    <Link
      to="/item/undefined"
    />
  </h2>
  <button
    className="Item_delete"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
  </button>
</div>
`;

exports[`Item component renders the Item given props 1`] = `
<div
  className="Item"
>
  <h2
    className="Item_title"
  >
    <Link
      to="/item/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Item_delete"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
  </button>
</div>
`;
